source/kraken/kraken_zmq.h

```diff
@@ -105,7 +107,10 @@ inline void doWork(zmq::context_t& context,
         }
 
         api = pb_req.requested_api();
-        log4cplus::NDCContextCreator ndc(pb_req.request_id());
+        std::string request_id = pb_req.request_id();
```

Comment:
```suggestion
        const std::string& request_id = pb_req.request_id();
```
Unless you really want to work on a copy ?